io/net/ethtool_test.py

```diff
@@ -102,19 +117,42 @@ def test_ethtool(self):
                               ignore_status=True)
             if ret.exit_status != 0:
                 self.fail("failed")
-        if not self.ping_check('-c 5'):
-            self.fail("ping failed after interface is up")
+        if not HostInfo.ping_check(self, self.iface, self.peer, '10000',
+                                   flood=True):
+            self.fail("flood ping test failed")
+        if self.priv_test:
+            self.ethtool_toggle_priv_flags()
 
-    def ping_check(self, options):
+    def ethtool_toggle_priv_flags(self):
         '''
-        Checks if ping to peer works fine and returns True.
-        Returns False otherwise.
+        Toggle the priv flag settings of the driver.
         '''
-        cmd = "ping -I %s %s %s" % (self.iface, options, self.peer)
-        if process.system(cmd, shell=True, verbose=True,
-                          ignore_status=True) != 0:
-            return False
-        return True
+        priv_pass = []
+        priv_fail = []
+        for oper in ('toggle', 'setback'):
+            for line in self.ret_val.stdout_text.splitlines():
+                if "off" in line:
+                    val = "on"
+                else:
+                    val = "off"
+                if "flags" not in line:
+                    priv_flag = line.split(':')[0]
+                    cmd = "ethtool --set-priv-flags %s \"%s\" %s" % \
+                          (self.iface, priv_flag.rstrip(), val)
+                    ret1 = process.run(cmd, shell=True, verbose=True,
+                                       ignore_status=True)
+                    if ret1.exit_status == 0 or 'supported' in \
+                       ret1.stderr_text:
+                        priv_pass.append(priv_flag.rstrip())
+                    else:
+                        priv_fail.append(priv_flag.rstrip())
+            if not HostInfo.ping_check(self, self.iface,
+                                       self.peer, '500000', flood=True):
+                self.fail("Ping failed for %s oper = %s" %
+                          (",".join(list(set(priv_fail))), oper))
```

Comment:
@manvanthar My mistake. I had thought the fail was for individual private flags, but it is not. Please go ahead with your older implementation for this fail, and the code is ready to merge.

Reply:
Done changes are reverted!!

Reply:
@manvanthar thanks